pkg/daemon/controller/controller.go

```diff
@@ -105,6 +109,8 @@ type ServiceController struct {
 	watchNamespace string
 	psCmdRunner    powershell.CommandRunner
 	ctrl           controller.Controller
+	drainer        *drain.Helper
```

Comment:
Does this need to be part of the struct? You could instantiate in the reconcile function and use it there.

Reply:
Either the drainer or the client required to instantiate it needs to be part of the struct.

Reply:
I am fine keeping the drainer here then